### ISSUE TITLE
`get_predicted` confidence intervals in mixed-effects models

### ISSUE THREAD

**Issue by vincentarelbundock** [easystats/insight#677]
At the moment, confidence intervals around predicted values for `lmerMod` only take into account the fixed effects.

For example, in a model with Chick-level random intercepts and coefficients, each Chick will have intervals of the exact same width at every time point:

``` r
library(lme4)
library(insight)

fit1 <- lmer(
  weight ~ 1 + Time + (1 + Time | Chick),
  data = ChickWeight)

insight::get_predicted(fit1, ci = .95) |> 
    cbind(ChickWeight) |>
    within({CI_width = CI_high - CI_low}) |>
    subset(Time == 4, select = "CI_width") |>
    table()
#> CI_width
#>  4.1810057846375 4.18100578463751 
#>                5               44
```

I was chatting with @ASKurz and we were wondering if this is what most users would expect/want. In particular, it seems like many would want the random components to be accounted for in the computation. Should we supply CIs at all, issue a warning, or is the status quo just fine?

Curious what everyone things (and maybe especially @bwiernik )

**Comment by ASKurz** [comment]
Speaking for myself, the current behavior is not what I would have hoped for and I like the options Vincent suggested.

**Comment by bwiernik** [comment]
Definitely would prefer for the uncertainty due to random effects to be included. But I think this might be a limitation of lme4? I think maybe if we did bootstrap intervals it might work?

**Comment by bwiernik** [comment]
Because the clever linear algebra tricks that lme4 uses, I don't think it's feasible to get correlations between fix effect and random effect estimates in this package. (They could be obtained in glmmTMB but currently aren't provided because they are computationally expensive). 

We should probably add a message saying that these default intervals don't include uncertainty due to random effects. And then we can recommend using `ci_method = "boot"` as an alternative?

**Comment by ASKurz** [comment]
How computationally expensive are we talking in the glmmTMB case?

**Comment by strengejacke** [comment]
One possible way is following Ben's suggestion of adding sigma/random effect variances before computing SE for predictions:

https://bbolker.github.io/mixedmodels-misc/glmmFAQ.html#predictions-andor-confidence-or-prediction-intervals-on-predictions

**Comment by strengejacke** [comment]
Or we look at `merTools::predictInterval()`, but I think it's limited to few families only?

**Comment by vincentarelbundock** [comment]
Re: bootstrap

I played around with some ideas and didn't come out reassured. My guess is that a good handling of the nested structure of the data will require something fancy. We probably don't want to give the user the impression that a plain-vanilla approach will give great results by giving them a `ci_method`.

Re: Bolker

He sounds lukewarm about all of these solutions, and ends up recommending Bayes. Seems like a lot of work to implement for a suboptimal result... might be best to just remove CIs or issue a warning.

Re: `predictInterval`

Those are prediction intervals, not confidence, right?

**Comment by bwiernik** [comment]
@ASKurz see https://github.com/glmmTMB/glmmTMB/issues/691

**Comment by strengejacke** [comment]
> Re: bootstrap
> 
> I played around with some ideas and didn't come out reassured. My guess is that a good handling of the nested structure of the data will require something fancy. We probably don't want to give the user the impression that a plain-vanilla approach will give great results by giving them a `ci_method`.

I _think_ `lme4::bootMer()` takes RE structures into account.

> Re: Bolker
> 
> He sounds lukewarm about all of these solutions, and ends up recommending Bayes. Seems like a lot of work to implement for a suboptimal result... might be best to just remove CIs or issue a warning.
> 
> Re: `predictInterval`
> 
> Those are prediction intervals, not confidence, right?

Ok, you were thinking about some [bias adjustment](https://github.com/easystats/modelbased/issues/57#issuecomment-632588471)?

**Comment by bwiernik** [comment]
Yep, you're right. `lme4::bootMer()` should give what we are looking for

**Comment by bbolker** [comment]
See https://github.com/lme4/lme4/issues/739
